♻️ Reworked Trademark Policy [due Oct 10] (#195)
commonhaus/foundation: policies/trademark-policy.md

### Uses we consider non-infringing

#### Distribution of unmodified source code or artifacts packaged by us

When you redistribute an unmodified copy of our software, either in source code form or as packaged artifacts (such as executables, libraries, or jar files), you are not changing the quality or nature of the software.

In this case, you may retain the Word Marks and the Logos we have placed on the software to identify your redistribution.
This kind of use only applies if you are redistributing an official distribution from this Project that has not been changed in any way.

You can find files for the Logos and approved designs for packaging on our [Brand Standards][] page.

> [!NOTE]
>
> When we refer to "our software," we mean the software produced by CF Projects.
> "Artifacts packaged by us" refers to software artifacts packaged and published by each CF Project, using their own processes and infrastructure.

> #### Distribution of modified source code or artifacts packaged you

Reviewer: ```suggestion
#### Distribution of modified source code or artifacts packaged by you
```

This kind of use only applies if you are redistributing an official distribution from this Project that has not been changed in any way.

You can find files for the Logos and approved designs for packaging on our [Brand Standards][] page.

> [!NOTE]
>
> When we refer to "our software," we mean the software produced by CF Projects.
> "Artifacts packaged by us" refers to software artifacts packaged and published by each CF Project, using their own processes and infrastructure.

#### Distribution of modified source code or artifacts packaged you

If you compile and distribute bitwise identical packaged artifacts from our unmodified source code, you may retain our Word Marks and Logos.

You may use the Word Marks, but not the Logos, to truthfully describe the origin of the software that you are providing, that is, that the code you are distributing is a modification of our software. You may also say, for example, that "this software is derived from the source code for Hibernate version 5.2."

If your modification includes minor changes like bug fixes that preserve compatibility with our published artifacts, your artifacts should be labeled to indicate they were built by you.

> For instance, the Maven dependency io.quarkus:quarkus-arc:3.8.5.redhat-00004 clearly indicates it was built by Red Hat.

Reviewer 2: ```suggestion
For instance, the Maven dependency io.quarkus:quarkus-arc:3.8.5.redhat-00004 clearly indicates it was built by Red Hat.
```

I get where this is coming from but I don't think it can be put in exactly like this; because it depends on the packaging system and mechanism applied if that name matters.

i.e. in go-lang symbolic names are used and one way for a company red hat or otherwise would be forced to provide a supported build would be to use the same physical name to distribute under but from a different repository.  Thus the source you get it from will have a different name (i.e. maven.repository.redhat.com).

Author: removed the example. Left with generic "labeled"

This Policy is not intended to limit the lawful use of our trademarks but to clarify what we consider lawful use to be.
Trademark law can be ambiguous, so we aim to provide enough clarity for you to understand whether we will consider your use acceptable, either because it is licensed or it is non-infringing.

Trademarks assure that products meet a certain standard of quality.
However, because open source licenses allow anyone to modify the software freely, we cannot guarantee that modified versions distributed by others under the same name will maintain that standard.
It is therefore important to follow these guidelines when using our trademarks to ensure that users know they are using or receiving software that meets the standards associated with the Commonhaus Foundation.

The sections that follow describe what trademarks are covered by this Policy and the uses of the trademarks that are allowed without additional permission from us.
If you want to use our trademarks in ways that are not described in this Policy, please [contact us](#contact-and-further-information).

> We have not approved any use that does not comply with this Policy or for which we have not provided separate written permission, so if your use is not covered here you must determine for yourself whether the use is lawful.

Author: Insert a blank line to make this easier to spot.

```suggestion

We have not approved any use that does not comply with this Policy or for which we have not provided separate written permission, so if your use is not covered here you must determine for yourself whether the use is lawful.
```

What you can say instead:

- You like the Hibernate software
- You participate in the Hibernate community
- You are providing an unmodified version of the Hibernate software
- You wrote a book describing how to use the Hibernate software.

This fundamental requirement, that it is always clear to people what they are getting and from whom, is reflected throughout this Policy.
It should also serve as your guide if you are not sure about how you are using the Marks.

Trademark law does not allow your use of names or trademarks that are too similar to ours.

- You may not use an obvious variation of any of our Marks or any phonetic equivalent, foreign language equivalent, takeoff, or abbreviation for a similar or compatible product or service.

- You may not use or register, in whole or in part, the Marks as part of your own trademark, service mark, domain name, company name, trade name, product name or service name.

In addition:

- You may not use the Marks in association with the use or distribution of software if you are also not in compliance with the copyright license for the software.

> - You agree that you will not acquire any rights in the Marks and that any goodwill generated by your use of the Marks inures solely to our benefit.

Reviewer 3: "any goodwill generated by your use of the Marks inures solely to our benefit"

why is this necessary? i can't guarantee whose benefit our use of the mark inures to.

Author: "goodwill" in this context has a legal meaning. I'm thinking of adding this as a footnote to help explain:

```suggestion
- You agree that you will not acquire any rights in the Marks and that any goodwill generated by your use of the Marks inures solely to our benefit.[^1]

[^1]: This clause prevents someone who uses the trademark from later claiming that, because their use of the mark built up goodwill (meaning the brand’s reputation and value) for them, they own the trademark. By including this statement, it makes clear that any goodwill generated from using the mark belongs to the Foundation (and the Project), not the user (licensee). It also serves as evidence that the user agreed not to claim ownership of the mark or its associated goodwill.
```

### Uses we consider non-infringing

> #### Distribution of unmodified source code or artifacts packaged by us

Reviewer 3: "us" who is us? moderne still continues to package artifacts. commonhaus isn't doing releases.

Author: I'll look at adding a comment to this section that "us" is relative.

Both the authoritative source for artifacts and the group responsible for building those artifacts will vary by project.

When you redistribute an unmodified copy of our software, either in source code form or as packaged artifacts (such as executables, libraries, or jar files), you are not changing the quality or nature of the software.

In this case, you may retain the Word Marks and the Logos we have placed on the software to identify your redistribution.
This kind of use only applies if you are redistributing an official distribution from this Project that has not been changed in any way.

You can find files for the Logos and approved designs for packaging on our [Brand Standards][] page.

> 

Author: @jkschneider Something like this, to clarify "us" in this case?

```suggestion
> [!NOTE]
>
> Each Commonhaus Project defines its own processes for packaging artifacts.
> When we refer to “our software,” we mean the software produced by CF Projects.
> “Artifacts packaged by us” generally refers to software artifacts packaged and published by each CF Project,
> though the specific processes for building and releasing artifacts may vary across Projects.

```

If your modification includes minor changes like bug fixes that preserve compatibility with our published artifacts, your artifacts should be labeled to indicate they were built by you.
For instance, the Maven dependency io.quarkus:quarkus-arc:3.8.5.redhat-00004 clearly indicates it was built by Red Hat.

Substantive modifications, on the other hand, should be treated as a fork. For example, where the functionality diverges, group or artifact identifiers should be changed to ensure users are not confused about which variant of the software they are getting and from whom.
You may place your own trademarks or logos on forked versions of the software, as you have become the origin of that version. In such cases, you should not use our Logos.

#### Statements about compatibility, interoperability or derivation

You may use the Word Marks, but not the Logos, to truthfully describe the relationship between your software and ours.

> Our Mark should be used after a verb or preposition that describes the relationship between your software and ours.

Reviewer 3: i would like to continue to be able to say "OpenRewrite by Moderne" because we pour basically all the energy into it -- at least until that isn't true. this seems to prohibit that.

Author: I believe this would be specific permission requested by Moderne, which would fall under the "separate written permission" category.

As I'm thinking about it, this would be part of the contribution agreement between us (CF) and you (Moderne).

#### Publications and presentations

You can use the Word Marks in book and article titles, and the Logo in illustrations within the document, as long as the use does not suggest that we have published, endorse, or agree with your work.

#### Events

You can use the signage in the [Brand Standards][] to promote the Foundation and/or its Projects at events.

### Uses for which we are granting a license

#### User groups

You can use the Word Marks as part of your user group name provided that:

- The main focus of the group is the Foundation, its Projects, or its software;
- Any software or services the group provides are without cost;
- The group does not make a profit;
- Any charge to attend meetings are to cover the cost of the venue, food and drink only.

Note that the general considerations for all uses, above, still apply.
Specifically, you may not use or register the Marks as part of your own trademark, service mark, domain name, company name, trade name, product name or service name.

#### Promotional goods

"Promotional goods" are non-software goods that use the Marks and that are intended to advertise, promote, or show membership in the Foundation or its Projects.

> You may make promotional goods for free giveaway at open source conferences and events using resources from our [Brand Standards][] site.

Reviewer 4: ```suggestion
You may make promotional goods for free giveaway at technology-related conferences and events using resources from our [Brand Standards][] site.
```

I don't think there's a good reason to limit this specifically to "open source conferences".

- Can you identify the product or service in question without using the trademark;
- Are you avoiding a likelihood of confusion in the way that you have used the trademark; and
- Have you used only as much as is necessary to identify the product or service.

With our "Foobar Certification" example above, the person offering the certification would be allowed to say, under the nominative fair use doctrine, that she is offering "Maude's Certification for Foobar software."

It is almost never the case that using a Logo will be a nominative fair use since it will be a rare case where the logo is needed for strictly informational purposes.

## Proper trademark use

These rules hold true for all trademarks, not just ours, so you should follow them for our Marks as well as anyone else's.

### Use of trademarks in text

- Always distinguish trademarks from surrounding text with at least initial capital letters or in all capital letters.
    - Unacceptable: hibernate
    - Acceptable: Hibernate, HIBERNATE

- Always use trademarks in their exact form with the correct spelling, neither abbreviated, hyphenated, or combined with any other word or words.

>     - Unacceptable: Open Rewrite, open rewrite, open-rewrite, openrewrite

Reviewer 3: Not be able to use 'openrewrite' feels unnecessarily prescriptive

Author: ```suggestion
    - Unacceptable: Open Rewrite, open rewrite, open-rewrite
```

    - Acceptable: OpenRewrite
    - See our [Trademark List][] to verify exactly how the marks should appear.

- Don't pluralize a trademark.
    - Unacceptable: I have seventeen JBangs running in my lab.
    - Acceptable: I have seventeen JBang catalogs running in my lab.

- Don't use "a" or "the" to refer to an instance of the trademark.
    - Unacceptable: I put a JBang on my mom's computer.
    - Acceptable: I put a JBang catalog on my mom's computer.

- Don't use a trademark as a possessive.
   The following noun should be used in possessive form or the sentence reworded so there is no possessive.
   An exception to this can be when CF is used to refer to the Foundation‘s name.

>     - Unacceptable: OpenRewrite's unique properties ...

Reviewer 3: Also overly prescriptive. I'm pretty sure we've written this in the possessive many times in written publications.

Author: These are general practices for preserving the value of the mark over time. 

This is best practice according to the [International Trademark Association](https://www.inta.org/fact-sheets/trademark-use/).

The guidelines we list in this section come from that publication, in essence, but are placed in context with our Projects and reference our list of trademarks. 

I will add a reference to those materials at the top of the section.

    - Acceptable: Unique capability of OpenRewrite™ ...
    - Acceptable: Commonhaus Foundations‘s IP Policy ...

- Don't use a trademark as a verb. Trademarks are products or services, never actions.
    - Unacceptable: I JReleasered my computer today!
    - Acceptable: I installed JReleaser on my computer today!

- Always use a trademark as an adjective modifying a noun. You can see the nouns we prefer on our [Trademark List][]
    - Unacceptable: This is an OpenRewrite. Anyone can use it.

>     - Acceptable: This is a OpenRewrite recipe. Anyone can use it.

Author: ```suggestion
    - Acceptable: This is an OpenRewrite recipe. Anyone can use it.
```

- Don't translate a trademark into another language.
    - Acceptable: Quiero instalar Hibernate en mi proyecto.
    - Unacceptable: Quiero instalar hibernar en mi proyecto.

### Use of Logos

> Use only official versions of the Logo obtained from the Commonhaus Foundation or its projects.

Reviewer 3: Unnecessarily prescriptive. For example, there is a 3d form of the OpenRewrite logo we use (and paid significantly) for that would now run afoul of this.

<img width="1014" alt="image" src="https://github.com/user-attachments/assets/f86580e0-a688-4c74-ba75-43f02e3c997f">

Author: In the general case, yes, that is exactly what we want to say is not permitted. 

What we should be doing in the specific case of Commonhaus, OpenRewrite, and Moderne is something else (per your point below).

Author: Again, I think this falls in the "specific written agreement between Commonhaus and Moderne" category.

Author: Capitalize Projects to recognize those as sub-entities.

```suggestion
Use only official versions of the Logo obtained from the Commonhaus Foundation or its Projects.
```

All Projects should ideally have an artwork listing somewhere (official Logos including light mode, dark mode, and monotone). 
This clause states folks should only choose Logos from that list, and shouldn't modify them.

It does not pose any restrictions on the Project for how logos are created or added to that list. OpenRewrite (as a Commonhaus Project in a future state) is in control of where those Logos come from, which would include those designed by Moderne.

> [!NOTE]
>
> For plugins, extensions, and other add-on software, the "likelihood of confusion" concern runs counter to how we speak. Let's use Quarkus extensions as an example.
>
> It may seem normal to say "Alice's Quarkus extension", but "Quarkus extension" can be understood two different ways. That phrase could be interpreted to mean either "Alice's extension for Quarkus" or "an extension for Alice's Quarkus". By using this form, you've introduced a possible misunderstanding about the source of Quarkus. The prepositional form, "Alice's extension for Quarkus" avoids that ambiguity.

#### Use of trademarks to show community affiliation

This section outlines the use of our Marks in community contexts, such as avatars, decorative skins, or themes. Guidelines for using our Marks on websites are addressed separately below.

You are welcome to use our Word Marks and Logos in non-commercial, decorative contexts—such as avatars, themes, or personas—to show your support for the Project, provided that your usage does not imply official representation, endorsement, or affiliation with the Foundation, Project, or community.

#### Websites

> You may use the Word Marks and Logos, but not the Trade Dress, on your webpage to show your support for the Foundation or its Projects as long as:

Reviewer 5: Can we link to a definition of "Trade Dress" here, as it's not a term I'd heard of previously

Author: ```suggestion
You may use the Word Marks and Logos, but not the [Trade Dress](https://www.inta.org/topics/trade-dress/), on your webpage to show your support for the Foundation or its Projects as long as:
```
